dressForTheWeather/View Models/RecommendationViewModel.swift

```diff
@@ -86,15 +77,56 @@ final class RecommendationViewModel {
         )
     }
     
-    private func generateRecommendation(for lowTemp: Double, highTemp: Double, from items: [ClothingItem]) -> [ClothingItem] {
+    private func selectClothingItems(for lowTemp: Double, highTemp: Double, from items: [ClothingItem]) -> [ClothingItem] {
         return items.filter { $0.tempRange.contains(lowTemp) || $0.tempRange.contains(highTemp) }
     }
     
     private func setRecommendations(for lowTemp: Double, highTemp: Double) {
-        let recommendedItems = generateRecommendation(for: lowTemp, highTemp: highTemp, from: allClothingItems)
-        recommendedClothingItems = recommendedItems
-        let outfit = Outfit(components: recommendedItems)
-        recommendations = outfit.recommendations
+        let recommendedItems = selectClothingItems(for: lowTemp, highTemp: highTemp, from: allClothingItems)
+        
+        recommendedClothingItems = []
+        
+        BodyPlacement.allCases.forEach { currentPlacement in
+            let itemsWithPlacement = recommendedItems.filter { $0.placement.contains(currentPlacement) }
+            
+            guard let randomItem = itemsWithPlacement.randomElement(),
+                recommendedClothingItems.filter({ $0.placement.contains(currentPlacement) }).isEmpty else { return }
+            
+            recommendedClothingItems.append(randomItem)
+        }
+        
+        let outfit = Outfit(components: recommendedClothingItems)
+        recommendations = outfit.recommendationString
+    }
+    
+    public func updateBackgroundColors() -> [CGColor]? {
```

Comment:
Maybe this updateBackgroundColors function could just become a computed property like this:

```
var gradientColors: [CGColor] {
        let temperatureRange = TemperatureRange.from(temp: highTemp)
        
        guard let temperatureRangeColor = UIColor(named: temperatureRange.rawValue)?.cgColor,
              let warmerTemperatureRangeColor = UIColor(named: temperatureRange.secondaryTemperatureRange.rawValue)?.cgColor else { return [] }
        
        return [temperatureRangeColor, warmerTemperatureRangeColor]
    }
```
There may be a cleaner way to make the cgColor array from UIColors.... perhaps a mapping solution or something...

Reply:
I like making this a var. Implemented in d935fcef46644f02812aa451e2d9313f9c71548f

I'm going to live the cgColor thing the way it is for now, might be something to re-explore in the future.